org.scala-ide.sdt.core/src/org/scalaide/core/internal/ScalaPlugin.scala

```diff
@@ -53,29 +53,27 @@ object ScalaPlugin {
 }
 
 class ScalaPlugin extends IScalaPlugin with PluginLogConfigurator with IResourceChangeListener with IElementChangedListener with HasLogger {
-  import CompilerUtils.ShortScalaVersion
-  import CompilerUtils.isBinaryPrevious
-  import CompilerUtils.isBinarySame
 
-  import org.scalaide.core.SdtConstants._
-
- /** Check if the given version is compatible with the current plug-in version.
-   *  Check on the major/minor number, discard the maintenance number.
-   *
-   *  For example 2.9.1 and 2.9.2-SNAPSHOT are compatible versions whereas
-   *  2.8.1 and 2.9.0 aren't.
-   */
-  def isCompatibleVersion(version: ScalaVersion, project: ScalaProject): Boolean = {
-    if (project.isUsingCompatibilityMode())
+ /**
+  * Check if the given version is compatible with the current plug-in version.
+  * Check on the major/minor number, discard the maintenance number.
+  *
+  * For example 2.9.1 and 2.9.2-SNAPSHOT are compatible versions whereas
+  * 2.8.1 and 2.9.0 aren't.
+  */
+  def isCompatibleVersion(version: ScalaVersion, project: ScalaProject): Boolean = project.getCompatibilityMode match {
+    case Same ⇒
+      isBinarySame(ScalaVersion.current, version) // don't treat 2 unknown versions as equal
+    case Previous ⇒
       isBinaryPrevious(ScalaVersion.current, version)
-    else
-      isBinarySame(ScalaVersion.current, version)// don't treat 2 unknown versions as equal
+    case Subsequent ⇒
```

Comment:
That could be done, but I'm not sure if it is necessary to further subdivide the `Previous` and `Subsequent` part. 2.9 is as 2.10 previous and could therefore handled by the same logic. I thought about potential dotty support and it could be added by just extending the trait `CompatiblityMode`.